### PR TITLE
Adding feature to push and pull images with in-toto metadata

### PR DESCRIPTION
**Pull request by seb-bah**
This pull request adds functionality to pull/push docker images digest SHA256 alongside in-toto metadata to TUF. When I found signy it was the only way to associate bundles or images with in-toto metadata. Unfortunately, that required using porter to create a bundle, even if you didn't need the full functionality of a bundle. This PR combined features of docker push and docker pull and combines it with signy sign and signy verify. While not part of the CNAB Security Spec, this will help users such as me who were searching for similar functionality.

Usage:

signy --tlscacert root-ca.crt image push -i [image]

This command is nearly identical to the docker CLI command docker push when the environment variable DOCKER_CONTENT_TRUST=1 and DOCKER_CONTENT_TRUST_SERVER=[server:4443] are set. In addition to signing the digest, we additionally push the in-toto metadata to the trust server just like signy sign does.

To pull and image and verify it's digest SHA:

signy --tlscacert root-ca.crt image pull -i [image]

This will pull the image from the registry, verify it's digest against what is stored in notary, and verify it's in-toto metadata that was pulled down from TUF.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +178,18 @@ Notes:
 
 - see current limitations about the in-toto signing key of the root layout
 
+### To sign container images and put the info in TUF alongside it's in-toto metadata
+
+`signy --tlscacert root-ca.crt push -i [image]`
+
+This command is nearly identical to the docker CLI command `docker push` when the environment variable `DOCKER_CONTENT_TRUST=1` and `DOCKER_CONTENT_TRUST_SERVER=[server:4443]` are set. In addition to signing the digest, we require in-toto metadata to push to the trust server. Unlike `signy sign` where the in-toto metadata is optional, it is required for `signy push`.
+
+To pull an image and verify it's digest SHA and in-toto metadata:
```
Comment (radu-matei) [inline README.md]: nit: _its_

File: cmd/push.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+func newPushCmd() *cobra.Command {
+	const pushDesc = `
+Push to docker and notary with trust data`
+	push := pushCmd{}
+	cmd := &cobra.Command{
+		Use:   "push [target reference]",
+		Short: "Push",
+		Long:  pushDesc,
+		//Args:  cobra.ExactArgs(0),
```
Comment (radu-matei) [inline cmd/push.go]: Are the comments here still needed?

Comment (scottbuckel) [reply cmd/push.go]: Resolved.

File: README.md (hunk 1 above)
Comment (scottbuckel) [reply README.md]: Resolved in the next push.

**Comment by scottbuckel** [comment]
@radu-matei This is good to go. After we push this through I can create Issue tickets for the TODO items that I mentioned in the readme.

**Review by radu-matei — APPROVED** [review]
LGTM, thanks for the contribution!

Could you please rebase on top of / merge the main branch, I just merged #91 which should fix the CI failure.

Thanks!

**Comment by radu-matei** [comment]
One more thing, @scottbuckel, while you are online - could you please rebase and only keep the useful commits here?
I wouldn't want to squash all of them into a single commit.

And thanks a lot for your patience with this PR.

**Comment by scottbuckel** [comment]
@radu-matei If the issue is that there are too many commits then I don't have an issue with a squash merge. Are there particular commits that you would like me to remove? I think they're all technically 'useful'.

**Pull request merged**